compat/mingw.c

```diff
@@ -741,6 +741,48 @@ static int is_local_named_pipe_path(const char *filename)
 		filename[9]);
 }
 
+static int is_local_disk_file(const char *filename)
+{
+	wchar_t wpath[MAX_LONG_PATH];
+	wchar_t wfullpath[MAX_LONG_PATH];
+	size_t windex;
+
+	if (is_local_named_pipe_path(filename))
+		return 0;
+
+	/*
+	 * Do everything in wide chars because the drive letter might be
+	 * a multi-byte sequence.  See win32_has_dos_drive_prefix().
+	 */
+	if (xutftowcs_long_path(wpath, filename) < 0)
+		return 0;
+
+	/*
+	 * GetDriveTypeW() requires a final slash.
+	 */
+	windex = wcslen(wpath) - 1;
+	while (windex > 0) {
+		if (is_dir_sep(wpath[windex]))
+			break;
+		windex--;
+	}
+	wpath[windex++] = L'\\';
+	wpath[windex] = 0;
+
+	/*
+	 * Normalize the path.  If nothing else, this converts forward
+	 * slashes to backslashes.  This is essential to get GetDriveTypeW()
+	 * correctly handle some UNC "\\server\share\..." paths.
+	 */
+	if (!GetFullPathNameW(wpath, MAX_LONG_PATH, wfullpath, NULL))
```

Comment:
This is a pretty costly call. It would be better to use something like `convert_slashes()`.

Reply:
help i dont know what to do

Reply:
> This is a pretty costly call. It would be better to use something like `convert_slashes()`.

yeah, that is true, but `GetDriveTypeW` needs full path as its input. 
Currently, I use `GetDriveType` to check if a file path is local or not and I failed to find a solid better solution, do you have any suggestion? 
by the way, `has_dos_drive_prefix` should not work, for file may locate in a mapped network drive, such as drive "Z" mapped to a network folder.